app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt

```diff
@@ -170,19 +169,20 @@ internal class MessagesActivity :
     }
 
     private fun refreshAll() {
-        try {
-            viewModel.coilHandler.evict()
-        } catch (e: IOException) {
-            Logger.error(e, "Problem evicting Coil cache")
-        }
+        viewModel.coilHandler.evict()
         startActivity(Intent(this, InitializationActivity::class.java))
         finish()
     }
 
     private fun onRefresh() {
+        viewModel.coilHandler.evict()
         viewModel.messages.clear()
         launchCoroutine {
-            loadMore(viewModel.appId)
+            loadMore(viewModel.appId).forEachIndexed { index, message ->
+                if (message.image != null) {
+                    listMessageAdapter.notifyItemChanged(index)
+                }
+            }
```

Comment:
@jmattheis I let the `loadMore` method return the messages so it's possible to filter them.  
So only the messages containing an image should be refreshed now.

Reply:
Nice idea. The code looks good, I'll test this on the weekend.